### PR TITLE
[CIS-1003] Fix multiple messages in a channel cannot quote a single message

### PR DESCRIPTION
**Pull request by SagarSDagdu**
### What does this PR do
- This  PR fixes the issue where multiple messages in a channel could not quote a single same message.

### CODE REVIEW

**Review by VojtaStavik — APPROVED** [review]
Nice catch 👍

**Comment by VojtaStavik** [comment]
Don't forget CHANGELOG entry for this fix

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift [hunk 1]
```diff
@@ -1038,6 +1038,91 @@ class MessageDTO_Tests: XCTestCase {
...
+                database.viewContext.message(id: messageId)?.asModel()
+            )
+            loadedMessages.append(loadedMessage)
+        }
+        
+        XCTAssertEqual(createdMessages.count, loadedMessages.count)
+        XCTAssertEqual(loadedMessages.first?.quotedMessage, nil) // The very first message doesn't quote any message
+        
+        XCTAssertEqual(loadedMessages[1].quotedMessage?.id, createdMessages[0].id) // The second message quotes the first message
+        XCTAssertEqual(loadedMessages[1].quotedMessage?.author.id, loadedMessages[0].author.id)
```
Comment (evsaev) [inline Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift:1118]: No need to check other info but the quoted message, I think. Correctness of `DTO -> model` has covered by `test_messagePayload_asModel`

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift [hunk 2]
```diff
@@ -1038,6 +1038,91 @@ class MessageDTO_Tests: XCTestCase {
...
+        try [firstMessageId, secondMessageId, thirdMessageId].forEach { messageId in
+            // Load the messages one by one from the db and save them
+            let loadedMessage: ChatMessage = try XCTUnwrap(
+                database.viewContext.message(id: messageId)?.asModel()
+            )
+            loadedMessages.append(loadedMessage)
+        }
+        
+        XCTAssertEqual(createdMessages.count, loadedMessages.count)
+        XCTAssertEqual(loadedMessages.first?.quotedMessage, nil) // The very first message doesn't quote any message
```
Comment (evsaev) [inline Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift:1115]: nit: in tests we usually put comments above
```suggestion
        // The very first message doesn't quote any message
        XCTAssertEqual(loadedMessages.first?.quotedMessage, nil)
```

**Review by evsaev — APPROVED** [review]
LGTM 👍

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift (hunk 1 above)
Comment (SagarSDagdu) [reply Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift]: Removed the unnecessary checks 👍

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift (hunk 2 above)
Comment (SagarSDagdu) [reply Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift]: Changed to match our code style

**Pull request merged**